src/main.c

```diff
@@ -45,33 +45,29 @@ int main(int argc, char **argv) {
         return 1;
     }
 
-    InitErrorBuffers();
+    InitCompiler();
 
-    const char *path = argv[1];
-    const char *data = ReadFile(path);
-    if (!data) {
-        perror("ReadFile");
-        exit(1);
-    }
-    Lexer lexer = MakeLexer(data, path);
-
-    Token token;
-    while ((token = NextToken(&lexer)).kind != TK_Eof) {
-        switch (token.kind) {
-            case TK_Int:
-                printf("kind: '%s', lit: '%llu'\n", DescribeTokenKind(token.kind), token.val.i);
-                break;
-            case TK_Float:
-                printf("kind: '%s', lit: '%f'\n", DescribeTokenKind(token.kind), token.val.f);
-                break;
-            case TK_String: case TK_Ident: case TK_Keyword: case TK_Directive:
-                printf("kind: '%s', lit '%s'\n", DescribeTokenKind(token.kind), token.val.s);
-            default:
-                printf("kind: '%s'\n", DescribeTokenKind(token.kind));
+    // FIXME: We need to work out what argv refers to the file
+    Package *mainPackage = ImportPackage(argv[1]);
+
+    u64 tokCount = 0;
+    while (QueueDequeue(&parsingQueue)) {
+        void *mem = ReadFile(mainPackage->fullPath);
+        if (!mem) {
+            printf("Failed to open file %s\n", mainPackage->fullPath);
```

Comment:
We should probably take the time to go find and abstract out all of our printf calls into our own printing utility. That way it should be easier to make things more testable. Not really about this PR, I'm just wanting to put it here so I don't forget.